src/BTVNanoCommissioning/utils/correction.py

```diff
@@ -0,0 +1,121 @@
+import importlib.resources
+import gzip
+import pickle
+import contextlib
+import cloudpickle
+
+import numpy as np
+import awkward as ak
+from coffea.lookup_tools.lookup_base import lookup_base
+from coffea.lookup_tools import extractor
+
+
+from coffea.lumi_tools import LumiMask
+from coffea.btag_tools import BTagScaleFactor
+from coffea.jetmet_tools import JECStack, CorrectedJetsFactory
+from coffea.lookup_tools import extractor
+
+# FIXME make jsons campaing configurable
+lumiMasks = {}
+_lumi_path = "BTVNanoCommissioning.data.lumiMasks"
+with importlib.resources.path(_lumi_path, "Cert_271036-284044_13TeV_23Sep2016ReReco_Collisions16_JSON.txt") as filename:
+    lumiMasks["2016"] = LumiMask(filename)
+with importlib.resources.path(_lumi_path, "Cert_294927-306462_13TeV_EOY2017ReReco_Collisions17_JSON_v1.txt") as filename:
+    lumiMasks["2017"] = LumiMask(filename)
+with importlib.resources.path(_lumi_path, "Cert_314472-325175_13TeV_17SeptEarlyReReco2018ABC_PromptEraD_Collisions18_JSON.txt") as filename:
+    lumiMasks["2018"] = LumiMask(filename)
+
+##JEC
+def load_jetfactory(campaign,path):
+    _jet_path =f"BTVNanoCommissioning.data.JME.{campaign}"
+    with importlib.resources.path(_jet_path, path) as filename:
+        with gzip.open(filename) as fin:
+            jmestuff = cloudpickle.load(fin)
+
+    jet_factory = jmestuff["jet_factory"]
+    return jet_factory
+
+def add_jec_variables(jets, event_rho):
+    jets["pt_raw"] = (1 - jets.rawFactor)*jets.pt
+    jets["mass_raw"] = (1 - jets.rawFactor)*jets.mass
+    if hasattr(jets, "genJetIdxG"):jets["pt_gen"] = ak.values_astype(ak.fill_none(jets.matched_gen.pt, 0), np.float32)
+    jets["event_rho"] = ak.broadcast_arrays(event_rho, jets.pt)[0]
+    return jets
+
+## PU weight
+def load_pu(campaign,path):
+    _pu_path =f"BTVNanoCommissioning.data.PU.{campaign}"
+    with importlib.resources.path(_pu_path, path) as filename:
+        with gzip.open(filename) as fin:
+            compiled = cloudpickle.load(fin)
+    return compiled
+
+## BTag SFs
+def load_BTV(campaign,path):
+    _btag_path = f"BTVNanoCommissioning.data.BTV.{campaign}"
+    
+    with importlib.resources.path(_btag_path,path['DeepCSVB']) as filename:
+        deepcsvb_sf = BTagScaleFactor(filename, BTagScaleFactor.RESHAPE,
+                                    methods='iterativefit,iterativefit,iterativefit')
+    with importlib.resources.path(_btag_path, path['DeepJetB']) as filename:
+        deepjetb_sf = BTagScaleFactor(filename, BTagScaleFactor.RESHAPE,
+                                    methods='iterativefit,iterativefit,iterativefit')
+
+    deepcsvc_sf = "BTVNanoCommissioning/data/BTV/"+campaign+"/"+path['DeepCSVC']
```

Comment:
does this work if submitted to condor?

Reply:
Yap, I check with `parsl/condor` in desy and it works